drop-transfer/src/ws/client/v2.rs

```diff
@@ -96,6 +96,60 @@ impl<const PING: bool> HandlerLoop<'_, PING> {
         Ok(())
     }
 
+    async fn issue_reject(
+        &mut self,
+        socket: &mut WebSocket,
+        file_id: FileId,
+    ) -> anyhow::Result<()> {
+        let file_subpath = if let Some(file) = self.xfer.files().get(&file_id) {
+            file.subpath().clone()
+        } else {
+            warn!(self.logger, "Missing file with ID: {file_id:?}");
+            return Ok(());
+        };
+
+        let msg = v2::ClientMsg::Cancel(v2::Download {
+            file: file_subpath.clone(),
+        });
+        socket.send(Message::from(&msg)).await?;
+
+        if let Some(task) = self.tasks.remove(&file_subpath) {
+            if !task.job.is_finished() {
+                task.job.abort();
+
+                task.events
+                    .stop(crate::Event::FileUploadCancelled(
+                        self.xfer.clone(),
+                        file_id.clone(),
+                        false,
+                    ))
+                    .await;
```

Comment:
It's confusing to see which version works where. This is `v2` file so I would assume this is v2 logic in here however it's clearly v5 logic inside of this file.

Another point - rejection fails if the cancel fails. I think rejection should follow through fully locally despite any errors on the other end. This is only for this file as client/v4 and client/v5 have it without cancel.

it looks like this should be `on_reject` instead of `issue_reject`. TLDR: architecture is very confusing to an external reader, could you clarify things?

Reply:
Actually, V4 does not have rejections and it's the same logic.
I wrote in the opening comment that there are two approaches:
* fallback to cancel
* Return to the caller with the error

I picked the first one. Can switch to the second option quite painlessly

`on_x()` method generally contains the handler code for something that happened, e.g. when there is a download request you have the `on_download()` handler
`issue_x()` is when the caller requests something to happen, so, in this case, it is correctly named as `issue_reject()` since you cannot receive from peer rejection because the wire protocol does not support that

Reply:
I think I got it. `client/vN.rs` are client code when communicating with that version, for example, `client/v2.rs` might still be used if we communicate with v2 server? In that case it would make sense

Reply:
Yes, `client/v2.rs` is a client implementation for communicating with the V2 protocol. The same goes for the server.

So I presented in this PR the fallback for cancellation in the case of V2 and V4 protocols.
Do you think that's OK? We can switch to reporting errors to the caller, effectively disabling this functionality for communicating with legacy peers

Reply:
The conclusion is that for this moment we fallback to cancel